### DISCUSSION TITLE
Ensure resource does not exist

### DISCUSSION THREAD

**Discussion by SebastianJ91** [argoproj/argo-cd#17724] [Q&A]
Is there a way to ensure that a resource does NOT exist?

Our use case is to delete a network policy that is automatically created by a template in OpenShift when a new namespace is created.
The easiest solution would be to change the template, but this is not possible for us due to different responsibilities in the cluster. 

The workaround we know of would be to overwrite the network policy with another ineffective policy.

**Comment by jgwest** [comment]
Theoretically possible, but definitely not commonly done. What comes to mind is this:


Argo CD uses `app.kubernetes.io/instance: (name of argo cd annotation)` annotation to track which resources belong to which Argo CD Applications. (More on this annotation [here](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/))

So if you had the ability to add the above annotation to the resource you want to delete, and the annotation points to an Argo CD Application which does NOT have the resource defined in Git, it would cause the resource to be deleted.

Note this would erase ANYTHING in the template with this annotation: you are essentially telling Argo CD that it owns these resources (and their lifecycle), and Argo CD will detect that the resource is not defined in git, and thus prune it.

You would also need to enable prune to ensure the resources are removed:
```
spec:
  syncPolicy:
    automated:
      prune: true
```

So, theoretically possible, but it would require the ability to add an annotation to the resource generated by the template.
